spec/etl/ga/concerns/transform_path_spec.rb

```diff
@@ -11,9 +11,30 @@ class Dummy
     events_with_prefix = Events::GA.where("page_path ~ '^\/https:\/\/www.gov.uk'")
     expect(events_with_prefix.count).to eq 1
 
-    Dummy.new.remove_invalid_prefix
+    Dummy.new.format_events_with_invalid_prefix
 
     events_with_prefix = Events::GA.where("page_path ~ '^\/https:\/\/www.gov.uk'")
     expect(events_with_prefix.count).to eq 0
   end
+
+  context 'when an event exists with the same page_path after formatting' do
+    let!(:event2) { create(:ga_event, page_path: "/https://www.gov.uk/topics", process_name: 'views', unique_pageviews: 1, pageviews: 1) }
+
+    before(:each) do
+      create(:ga_event, page_path: "/topics", process_name: 'views', unique_pageviews: 100, pageviews: 200)
+      Dummy.new.format_events_with_invalid_prefix
+      event2.reload
```

Comment:
Could you please move lines 25 and 26 inside the tests. Duplication in testing in ok when exercising our code.

I would refactor into:

```ruby
  context 'when an event exists with the same page_path after formatting' do
     subject { Dummy.new }

     let!(:event2) { create(:ga_event, page_path: "/https://www.gov.uk/topics", process_name: 'views', unique_pageviews: 1, pageviews: 1) }

    before(:each) do
      create(:ga_event, page_path: "/topics", process_name: 'views', unique_pageviews: 100, pageviews: 200)
    end

    it 'updates events with their combined pageviews' do
      subject.format_events_with_invalid_prefix

      expect(event2.reload.pageviews).to eq 201
    end

    it 'updates events with their combined unique_pageviews' do
      subject.format_events_with_invalid_prefix

      expect(event2.reloadunique_pageviews).to eq 101
    end

    it 'deletes the duplicated event' do
      subject.format_events_with_invalid_prefix

      expect(Events::GA.count).to eq 1
    end
  end